fox_consumer/consumer.py

```diff
@@ -0,0 +1,361 @@
+                    "item_main_category": {"type": "string"},
+                    "item_type": {"type": "string"},
+                    "item_name": {"type": "string"},
+                    "item_price": {"type": "number"},
+                },
+                "required": ["item_img_id", "item_main_category", "item_type", "item_name", "item_price"]
+            }
+
+        self.consumer_db = FoxConsumerPostgres()
+        self.initialize_rabbitmq_connection()
+
+    def close_connections(self):
+        # TODO - Caduri - since it should run continuously, do we need this method? [TBD]
+        self.rabbitmq_connection.close()
+        self.consumer_db.close_connections()
+
+    def initialize_rabbitmq_connection(self):
+        """
+        This function will initialize rabbitmq connection.
+        The FoxConsumer will connect to a pre-define exchange server (fox_scrap) and start to wait for input
+        """
+        self.rabbitmq_connection = pika.BlockingConnection(pika.ConnectionParameters(host='localhost'))
+        self.rabbitmq_channel = self.rabbitmq_connection.channel()
+        self.rabbitmq_channel.exchange_declare(exchange='fox_scrap', type='fanout')
+        result = self.rabbitmq_channel.queue_declare(exclusive=True)
+        queue_name = result.method.queue
+        self.rabbitmq_channel.queue_bind(exchange='fox_scrap', queue=queue_name, routing_key='')
+        self.rabbitmq_channel.basic_consume(self.callback, queue=queue_name, no_ack=True)
+        self.rabbitmq_channel.start_consuming()
+
+    def callback(self, method, properties, asd, body):
+        """
+        This is the "main" function for the FoxConsumer.
+        every time the consumer recieve input it will verify it and add to the DB.
+        :param body: input from rabbitmq server
+        """
+        print(" [x] Received %r" % body)    # TODO - will be deleted at the end...
+
+        if self.verify_json(body):
+            current_item = self.convert_json_to_fox_item(body)
+            self.consumer_db.save_item(current_item)
+        else:
+            print "the input is corrupted..."
+
+    def verify_json(self, data):
+        """
+        Verify json input according to the schema defined for FoxItem class
+        :param data: json input from rabbitmq server
+        :return: True/False (verified/not)
+        """
+        try:
+            jsonschema.validate(json.loads(data), self.schema)
+            return True
+        except jsonschema.ValidationError as e:
+            print e.message
+        except jsonschema.SchemaError as e:
+            print e
+        return False
+
+    def convert_json_to_fox_item(self, obj):
+        """
+        This function will convert the json to FoxItem class
+        :param obj: json object
+        :return: FoxItem object
+        """
+        item = json.loads(obj)
+        item = FoxItem(item['item_img_id'], item['item_main_category'], item['item_type'], item['item_name'], item['item_price'])
```

Comment:
If `item` includes all the parameters for `FoxItem` in the same name(which is the case here) you can write `item = FoxItem(**item)`